src/server/rdb_save.cc

```diff
@@ -379,7 +381,7 @@ error_code RdbSerializer::SaveListObject(const robj* obj) {
         if (decompressed)
           zfree(decompressed);
       });
-      RETURN_ON_ERR(SaveListPackAsZiplist(lp));
```

Comment:
> but save everything in a format that Redis6 understands
I can revert the changes to what we serialized before, and this should do it. I can also create another issue about it. 

Also one more thing, I think this change pre-existed version 10 of rdb, which means that it should work with Redis6. I will confirm it and ping here.

Reply:
@romange Yes I am right, this was a bug on our end. See https://github.com/redis/redis/blob/6.2/src/rdb.c#L672C21-L672C21 which serializes to `RDB_TYPE_LIST_QUICKLIST`. `This is redis 6.2`. Therefore, I will downgrade to `QUICKLIST instead of QUICKLIST_2` to be compatible. On loading, we take care of `ziplists` separately so we should be ok.

Reply:
Also, even in `redis 5` this is true, see https://github.com/redis/redis/blob/5.0/src/rdb.c#L630C7-L630C7

Reply:
`redis 4` the same. Ziplists are `redis 3` lol :D See https://github.com/redis/redis/blob/3.0/src/rdb.c#L436

Reply:
@romange I deleted my comments, brainfarted heavily and I was looking at the wrong thing. Yes indeed, listpack encoding does not exist on redis 6.